models/nftb/bnb/nftb_bnb_events.sql

```diff
@@ -0,0 +1,298 @@
+        END  as seller
+            
+       ,'Trade' as evt_type
+       ,CAST(1 AS DECIMAL(38,0)) AS number_of_items
+       ,'Single Item Trade' as trade_type
+       ,CASE WHEN tr.from='0xebd4232e4c1999bc9562802eae01b431d5053e65'
+              THEN 'Auction Settled'
+              WHEN tr.from='0xdc2f08364ebc6cebe0b487fc47823b1e83ce8550'
+              THEN 'Claim'
+              ELSE 'Buy' END as trade_category
+       ,'BNB' as currency_symbol
+       ,'0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c' as currency_contract
+       ,'0x836eb8202d4bc2ed14d1d2861e441c69228155cc' AS nft_contract_address
+       ,CAST (element_at(r.output_0, 1) AS STRING) as royalty_fee_receive_address
+       ,CASE WHEN tr1.value=0 AND tra1.address='0xebd4232e4c1999bc9562802eae01b431d5053e65' --seller contract address with withdraw function is called (auction settled)
+             THEN tra1.value
+             WHEN tr1.value=0 AND tra1.address='0xdc2f08364ebc6cebe0b487fc47823b1e83ce8550' -- seller contract address with claim function is called
+             THEN tra1.value
+             ELSE tr1.value END AS amount_raw
+       ,CASE WHEN cardinality(r.output_1)>1 THEN CAST (10 AS DOUBLE)
+              WHEN cardinality(r.output_1)=1 THEN CAST (element_at(r.output_1, 1) AS DOUBLE)
+              ELSE CAST (0 AS DOUBLE)
+              END AS royalty_fee_percentage
+          
+FROM {{ source('nftb_bnb', 'NFT_evt_Transfer')}} tr
+INNER JOIN {{ source('nftb_bnb', 'NFT_call_royaltyInfo')}} r
+ON r.call_tx_hash=tr.evt_tx_hash
+INNER JOIN {{ source('bnb','transactions')}} as tr1
+ON tr1.hash=tr.evt_tx_hash
+{% if not is_incremental() %}
+AND tr1.block_time >= '{{project_start_date}}'
+{% endif %}
+{% if is_incremental() %}
+AND tr1.block_time >= date_trunc("day", now() - interval '1 week')
+{% endif %}
+LEFT JOIN (SELECT 
+                   bt.from as address
+                  ,tx_hash
+                  ,SUM(CAST (value AS DECIMAL(38,0))) AS value
+            FROM {{ source ('bnb','traces')}} bt
+            WHERE 1=1
+            AND (bt.from='0xebd4232e4c1999bc9562802eae01b431d5053e65' --seller contract address when withdraw function is called
+            OR
+            bt.from='0xdc2f08364ebc6cebe0b487fc47823b1e83ce8550'-- seller contract address when claim function is called
+            )
+            AND bt.input='0x'
+            AND bt.call_type='call'
+            {% if not is_incremental() %}
+            AND bt.block_time >= '{{project_start_date}}'
+            {% endif %}
+            {% if is_incremental() %}
+            AND bt.block_time >= date_trunc("day", now() - interval '1 week')
+            {% endif %}
+            GROUP BY 1,2
+            ) tra1 
+ON tra1.tx_hash=tr.evt_tx_hash
+LEFT JOIN {{ source('bnb','logs')}} l1 
+ON l1.tx_hash=r.call_tx_hash
+AND l1.contract_address='0xdc2f08364ebc6cebe0b487fc47823b1e83ce8550'
+AND l1.topic1='0x7a9edcf72fda2a090305f15fe2f1d8d881c849c4a142e8847734fe93542c64ef' --func sig for claim function call
```

Comment:
Is there a reason we are using the logs directly instead of decoding the contracts and using the decoded function call tables?

Reply:
the contract `0xdc2f08364ebc6cebe0b487fc47823b1e83ce8550`  and `0xe22c90e7816db4344f33c651c7b0a01fcd51a327` are both  **TransparentUpgradeableProxy**  contracts. Their ABIs are present while `0xebd4232e4c1999bc9562802eae01b431d5053e65` contract is not verified on etherscan. I have submitted the first two for decoding right now. I think i expected to see emitted events for the functions and on not seeing them decided to use logs to get the information.

Reply:
@LeviScoffie 
Ok looks like all the contracts are not verified which will make it impossible to get those proper call tables.
I assume you submitted the ABI from etherscan which is the Proxy ABI, and not the implementation ABI.

proxy -> implementation
[0xdc2f08364ebc6cebe0b487fc47823b1e83ce8550](https://bscscan.com/proxyContractChecker?a=0xdc2f08364ebc6cebe0b487fc47823b1e83ce8550) -> [0xec53739a072e1aa8cc2d9849e096d3b440bf3f3c](https://bscscan.com/address/0xec53739a072e1aa8cc2d9849e096d3b440bf3f3c#code) (:x: unverified)
[0xe22c90e7816db4344f33c651c7b0a01fcd51a327](https://bscscan.com/proxyContractChecker?a=0xe22c90e7816db4344f33c651c7b0a01fcd51a327) -> [0xe93055a69323253320150bfe1c40a18c493ab267](https://bscscan.com/address/0xe93055a69323253320150bfe1c40a18c493ab267#code) (:x: unverified)
[0xebd4232e4c1999bc9562802eae01b431d5053e65](https://bscscan.com/proxyContractChecker?a=0xebd4232e4c1999bc9562802eae01b431d5053e65) -> [0x8947bd49d43501e9872868b770074e10528c0389](https://bscscan.com/address/0x8947bd49d43501e9872868b770074e10528c0389#code) (:x: unverified)

We would need to get the ABI's from those implementation contracts to successfully decode the transactions.
If you had contact with NFTb I suggest reaching out and asking if they can supply the ABI's, if not our only option is to continue to work with the raw logs as you did.

Reply:
Let me reach out to the team. Hopefully to get a substansive response

Reply:
@0xRobin so apparently, they dont share that information with the public. I guess we'll just have to use the log tables

Reply:
ugh, even not the ABIs? That's a pitty.

Reply:
Yeah @0xRobin . Really sad. Would using the log tables bring up perfomance issues?

Reply:
It might be a bit less performant, but not that it should block this PR.